flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/dataset/DataSetCalc.scala

```diff
@@ -129,8 +131,35 @@ class DataSetCalc(
       body,
       returnType)
 
+    def getForwardIndices = {
+      //get indices of all modified operands
+      val modifiedOperands = calcProgram.
+        getExprList
+        .filter(_.isInstanceOf[RexCall])
+        .flatMap(_.asInstanceOf[RexCall].operands)
+        .map(_.asInstanceOf[RexLocalRef].getIndex)
```

Comment:
Shouldn't the modified fields are meaning to the input fields? So it should be RexInputRef?